src/scss/_functions.scss

```diff
@@ -116,31 +116,75 @@
+/// the same but the saturations and luminance changes.
+///
+/// Not all our colours allow tones. If a colour cannot be toned an error is
+/// thrown. You may however mix the colour with another supported colour.
+/// @see oColorsMix
 ///
 /// @param {String} $color-name - the name of the color to be shaded
-/// @param {Number} $brightness - the brightness value of the new color
-@function oColorsGetTint($color-name, $brightness) {
-	$hue: null;
-	$saturation: null;
-	$mix: null;
-	$newcolor: null;
-
-	@if not (map-has-key($_o-colors-tints, $color-name)) {
-		$color-tints: map-keys($_o-colors-tints);
-		@return _error("It is not possible to use a tint of: #{$color-name}, please use one of: #{$color-tints} instead.");
+/// @param {Number} $brightness - the brightness value of the new color, 0-100
+@function oColorsGetTone($color-name, $brightness) {
+	$color: oColorsByName($color-name);
+	$hue: hue($color);
+	// Find a tone from a base of 90% saturation.
+	// The reason for this is not known. It's here to preserve the tones we
+	// currently have. The base starting used to be configured per colour
+	// and may once have differed from colour to colour.
+	$saturation: 90;
+
+	// Validate the given colour allows different tones.
+	$has-tone-config: map-has-key($_o-colors-default-palette-tones, $color-name);
+	@if (not $has-tone-config) {
+		@return _error('"#{$color-name}" does not support tones. ' +
+			'Consider using a mix instead e.g: ' +
+			'`oColorsMix(\'#{$color-name}\', $percentage: #{$brightness})`');
+	}
+
+	// Validate brightness.
+	@if(type-of($brightness) != 'number' or $brightness > 100 or $brightness < 0) {
+		@return _error('"$brightness" must be a number between 0 and 100.');
 	}
 
-	@if (map-has-key(map-get($_o-colors-tints, $color-name), 'hue')) {
-		$hue: map-get(map-get($_o-colors-tints, $color-name), 'hue');
-		$saturation: map-get(map-get($_o-colors-tints, $color-name), 'saturation');
-		$newcolor: _oColorsHSB($hue, $saturation, $brightness);
+	@if $brightness == 0 {
+		@return hsla(0, 0, 0, 1);
 	} @else {
-		$newcolor: oColorsMix($color-name, $percentage: $brightness);
+		$hsl-luminance: ($brightness/2) * (2 - ($saturation/100));
+		$hsl-saturation: ($brightness * $saturation) / if($hsl-luminance < 50, $hsl-luminance * 2, 200 - $hsl-luminance * 2);
+
+		@return hsla($hue, $hsl-saturation, $hsl-luminance, 1);
 	}
+}
 
-	@return $newcolor;
+/// Figure out if a given colour is a tone. If it is a tone return the original
+/// colour name and its tone value, otherwise return null.
+///
+/// @example Get the tone value of 'claret-80'
+///    $tone-details: oColorsGetToneDetails('claret-80');
+///    $color: map-get($tone-details, 'color'); // claret
+///    $value: map-get($tone-details, 'value'); // 80
+///
+/// @param {String} $color - the palette colour hex or color name e.g. 'claret-80'
+/// @return {Map|Null} - the details of the given tone e.g. ('color': 'claret', 'value': 80)					)
```

Comment:
I guess there is a reason for this to return `null` if no tone is found but I can't seem to think of the situation where I'd want `null` instead of an error. Could you help me understand the situation @notlee?

Reply:
Yeah the situation is where you want to do something conditionally if a color is not a tone. We could error here and introduce another function `oColorsColourIsTone` but felt that that was a bit much considering its usecase.